src/main/java/com/github/rawsanj/handler/WebHttpHandler.java

```diff
@@ -23,7 +24,9 @@ public RouterFunction<ServerResponse> htmlRouter(@Value("classpath:/static/index
 		return route(GET("/"), request -> ok().contentType(MediaType.TEXT_HTML).bodyValue(html))
 			.andRoute(POST("/message"), request -> request.bodyToMono(Message.class)
 				.flatMap(message -> redisChatMessagePublisher.publishChatMessage(message.getMessage()))
-				.flatMap(aLong -> ServerResponse.ok().bodyValue(new Message("Message Sent Successfully!."))));
+				.flatMap(aLong -> ServerResponse.ok().bodyValue(new Message("Message Sent Successfully!.")))
+					.subscribeOn(Schedulers.boundedElastic()));
```

Comment:
Do we really need to make this run on `Schedulers.boundedElastic()` Did the [BlockHound](https://github.com/reactor/BlockHound) detected blocking call over here or inside the `RedisChatMessagePublisher`?

Reply:
![image](https://github.com/RawSanj/spring-redis-websocket/assets/56495631/42fb8d72-0954-46b1-9358-4b70bd958385)
In the stack trace, we see that the error propoages to **RedisChatMessagePublisher** but was triggered by **WebHttpHandler**, line 26 right? :)

Reply:
> In the first stack trace (first screenshot in the post), we see that the error propagates to **RedisChatMessagePublisher** but was triggered by **WebHttpHandler**, line 26 right? :)

The other way round. The actual blocking call is at line - https://github.com/RawSanj/spring-redis-websocket/blob/master/src/main/java/com/github/rawsanj/messaging/RedisChatMessagePublisher.java#L35C11-L35C28 the Atomic Increment is blocking.

I think we should be good if we move that blocking call `Integer totalChatMessage = chatMessageCounter.incrementAndGet()` inside the `Mono.fromCallable()` and then add the `.subscribeOn(Schedulers.boundedElastic())` in the end.